Assets/Scripts/GameManager/IGameEntity.cs

```diff
@@ -0,0 +1,11 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+/// <summary>
+/// A GameEntity is an object that is a component of a gameobject placed in the world. Its data is provided by the server.
+/// </summary>
+public interface IGameEntity
+{
+    public void Setup(IGameEntityData data);
```

Comment:
I recommend adding generics `IGameEntity`.
C# has something like `T extends IGameEntityData`, right?
And then, each of the implementations can specify the exact subtype it wants.
The parameter of `Setup` would then simply be `T`.
This has the benefit that you won't need these ugly casts you resort to at the moment.

Reply:
Thank you for that hint! I was searching for a way to connect a GameEntity to its GameEntityData but could not find anything.

Reply:
However, this has the disadvantage that you always have to give the IGameEntityData type on every method relating GameEntities even if the data type is not relevant.
![Screenshot 2023-01-01 221232](https://user-images.githubusercontent.com/98211563/210184744-ac0968aa-d9cc-43ee-a82c-dacc448f27c2.png)